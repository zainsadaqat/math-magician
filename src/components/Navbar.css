.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1em;
  font-family: 'Montserrat', sans-serif;
}

.nav {
  display: flex;
}

.nav-item {
  padding: 0 1em;
  border-left: 3px solid rgb(100, 100, 100);
  font-weight: bolder;
}

.nav-item:first-child {
  border-left: none;
}

.nav-item > a {
  color: rgb(100, 100, 100);
}

.nav-item > a:hover {
  color: rgb(54, 54, 54);
  text-decoration: underline;
}

.active {
  color: #333;
}
