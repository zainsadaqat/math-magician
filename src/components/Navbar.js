import React from 'react';
import './Navbar.css';
import { Link } from 'react-router-dom';

const Navbar = () => {
  return (
    <header className="navbar">
      <h1>Math Magicians</h1>
      <nav>
        <ul className="nav">
          <li className="nav-item">
            <Link to="/">Home</Link>
          </li>
          <li className="nav-item">
            <Link to="/calculator">Calculator</Link>
          </li>
          <li className="nav-item">
            <Link to="/quote">Quote</Link>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Navbar;
