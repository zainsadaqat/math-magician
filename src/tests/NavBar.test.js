import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('Matches Snapshot', () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
