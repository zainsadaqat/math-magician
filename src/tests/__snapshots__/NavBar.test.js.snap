// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<header
  className="navbar"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      className="nav"
    >
      <li
        className="nav-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
